OpenSSL/test/test_crypto.py

```diff
@@ -3431,6 +3434,66 @@ def test_modification_pre_verify(self):
         self.assertEqual(store_ctx.verify_certificate(), None)
 
 
+    def _make_test_crl(self, issuer_cert, issuer_key, certs=()):
+        """
+        Create a CRL.
+
+        :param list[X509] certs: A list of certificates to revoke.
+        :return: :py:class:`CRL`
+        """
+        crl = CRL()
+        for cert in certs:
+            revoked = Revoked()
+            # FIXME: This string splicing is an unfortunate implementation detail
+            # that has been reported in
+            # https://github.com/pyca/pyopenssl/issues/258
+            serial = hex(cert.get_serial_number())[2:]
+            revoked.set_serial(serial)
+            revoked.set_reason(b'unspecified')
+            revoked.set_rev_date(b'20140601000000Z')
+            crl.add_revoked(revoked)
+        crl.set_version(1)
+        crl.set_lastUpdate(b'20140601000000Z')
+        crl.set_nextUpdate(b'20180601000000Z')
```

Comment:
I need to fix this to automatically detect the year, else we'll start failing sometime around 2018.

Sorry, but I can't resist:

![what-year-is-it](http://s2.quickmeme.com/img/98/98b39b21ff008afaa5cba1e79fc611b2ddb64b3f44f57e792397c517768f4433.jpg)